### ISSUE TITLE
Layouts with window navigation using incorrect color map for borders. 

### ISSUE THREAD

**Issue by nicholma** [xmonad/xmonad-contrib#138]
### Problem Description

When the layout hook includes WindowNavigation (eg `configurableNavigation noNavigateBorders`) borders are sometimes shown as transparent. 

Bellow is a recording of it.

Note I am using archlinux with the -git aur pages. 
The white terminal is termite, which shows the bug.
Other windows are gnome-terminal.

![6c38c026-df3e-11e6-890d-d984e0b686f5](https://cloud.githubusercontent.com/assets/1690826/22161943/5e2047c4-df44-11e6-8786-21138f5bf625.gif)

### Configuration File
```haskell
import XMonad
import XMonad.Config.Desktop
import XMonad.Layout.Spacing
import XMonad.Layout.ResizableTile
import XMonad.Layout.WindowNavigation

main :: IO ()
main = xmonad $ desktopConfig
    { terminal    = "gnome-terminal"
    , modMask     = mod4Mask
    , borderWidth = 30
    , layoutHook  = configurableNavigation noNavigateBorders $ spacing 12 $ ResizableTall 1 0.03 0.5 []
    }
```

### Checklist

  - [X] I've read [CONTRIBUTING.md](https://github.com/xmonad/xmonad/blob/master/CONTRIBUTING.md)

  - [ ] I tested my configuration with [xmonad-testing](https://github.com/xmonad/xmonad-testing)

**Comment by asjo** [comment]
This is fixed in the current development version of XMonad. The fix will be included in the next release. See https://github.com/xmonad/xmonad/pull/59

**Comment by geekosaur** [comment]
No, it is not. The border management in the core was fixed; X.L.WindowNavigation does its own border management, and that still uses the wrong colormap *and* completely ignores the alpha component (probably losing it completely, based on a quick read of the source) when computing border dimming.

**Comment by geekosaur** [comment]
There *may* be a similar bug in X.L.Decoration, although de facto its border settings get overwritten immediately by X.O.windows.

**Comment by asjo** [comment]
Ouch. Sorry for the misinformation on my part.

**Comment by pjones** [comment]
@geekosaur How do you feel about XMonad Core exporting a function for setting the border color of a window?  It will use the new logic we have in place and then we can update all -contrib modules that set the border.

**Comment by geekosaur** [comment]
I've actually been sitting on a combinator to simplify using X11 stuff directly from the X monad that needs some more thought, in part because of stuff like this --- but, arguably, things like border color and possibly border width probably ought to be exported from core for this reason.

**Comment by pjones** [comment]
@nicholma Would you please test #142.  That PR should fix your issue.

**Comment by pjones** [comment]
Fixed in 0.13
